### PR TITLE
Specify public key on ex_create_keypair and  Successfully get public ip ...

### PR DESCRIPTION
**Pull request by mgogoulos**
...addresses for nodes for HP Cloud

https://issues.apache.org/jira/browse/LIBCLOUD-449

### CODE REVIEW

File: libcloud/compute/drivers/openstack.py [hunk 1]
```diff
@@ -39,6 +39,7 @@
 from libcloud.common.openstack import OpenStackDriverMixin
 from libcloud.common.types import MalformedResponseError, ProviderError
 from libcloud.compute.types import NodeState, Provider
+from libcloud.compute.base import is_private_subnet
```
Comment (Kami) [inline libcloud/compute/drivers/openstack.py:42]: This function was recently moved to `libcloud.utils.networking` and a new `is_public_subnet` function was also added there.

File: libcloud/compute/drivers/openstack.py [hunk 2]
```diff
@@ -1730,7 +1731,7 @@ def ex_list_keypairs(self):
         return self._to_keypairs(
             self.connection.request('/os-keypairs').object)
 
-    def ex_create_keypair(self, name):
+    def ex_create_keypair(self, name, public_key=None):
```
Comment (Kami) [inline libcloud/compute/drivers/openstack.py:1734]: Please add a docstring for new `public_key` argument.

**Comment by Kami** [comment]
Those changes also need tests, especially the IP address change.

**Comment by Jc2k** [comment]
Why change ex_create_keypair? I think ex_import_keypair and/or ex_import_keypair_from_string do the same job already.

**Comment by Kami** [comment]
@Jc2k Ah, that's actually a good point & catch. For consistency, `ex_create_keypair` should only take a `name` argument, right?

I do need to admit, that I find the existing API slightly "weird", but it's what we have at the moment and we should follow it for the consistency reasons.

In any case, if we want to change this and get rid of the "import" methods, we should do it now, before the 1.0 release.

**Comment by mgogoulos** [comment]
closing, will open a new pull request for the second issue only

**Pull request closed without merging**